### PR TITLE
Fixed crash when generating keys in android with RTL locales

### PR DESCRIPTION
**Pull request by hmawla**
Created a workaround according to the community that would switch the locale to English before generating the keys, then switches back to the default one after that.

### CODE REVIEW

**Comment by hmawla** [comment]
#131

**Comment by mogol** [comment]
@hmawla good catch 👍 
Let me test and merge after.

**Comment by iassal** [comment]
@hmawla awesome stuff! 

@mogol when can this be released?

File: android/src/main/java/com/it_nomads/fluttersecurestorage/ciphers/RSACipher18Implementation.java [hunk 1]
```diff
@@ -168,6 +186,6 @@ private void createKeys(Context context) throws Exception {
             kpGenerator.initialize(spec);
             kpGenerator.generateKeyPair();
         }
+        setLocale(localeBeforeFakingEnglishLocale);
```
Comment (iassal) [inline android/src/main/java/com/it_nomads/fluttersecurestorage/ciphers/RSACipher18Implementation.java:189]: @mogol Just wondering if this should be encapsulated in a finally block instead? Would this have any side effects if an exception was thrown and this wasn't called?

**Comment by hmawla** [comment]
@iassal you can use my git if you need it urgently by adding [this ](https://pastebin.com/7TTcRLES)to your pubspec.yaml file
I won't be updating it furthermore until the next release is published on pub.dev.
Though make sure to follow @mogol when the next release is ready to switch back.

File: android/src/main/java/com/it_nomads/fluttersecurestorage/ciphers/RSACipher18Implementation.java (hunk 1 above)
Comment (hmawla) [reply android/src/main/java/com/it_nomads/fluttersecurestorage/ciphers/RSACipher18Implementation.java:189]: yes, it makes sense to do so just in case...

**Pull request merged**